lib/puma/server.rb

```diff
@@ -109,7 +109,29 @@ def initialize(app, events = nil, options = {})
       @precheck_closing = true
 
       @requests_count = 0
+
+      unless Client.const_defined? :ERROR_RESPONSE
```

Comment:
Rather than do this, would it be cleaner to modify the different call-sites using ERROR_RESPONSE to instead use a class method (maybe `Puma.error_responses`)? This process of defining a constant in response to the DSL is weird to me.

Reply:
Part of where it gets tricky is that `Client` and `Server`/`Request` aren't tightly bound, but I think `Server`/`Request` can call into `Client` methods, as it creates the `Client` instances. `Client` doesn't have any config/options info.

How we handle Puma generated responses could use some updating, as it's done a few different ways.

Kind of burnt, let me think about it.  Maybe we should just remove the Puma info (the `Server:` header), and take some more time to think about it...